dev-python/pymupdf/patches/pymupdf-1.19.6.patchset

```diff
@@ -0,0 +1,30 @@
+From fdefbb7634650726393dcae61c14842f8d3ea891 Mon Sep 17 00:00:00 2001
+From: Ivan Holmes <ivan@ivanholmes.co.uk>
+Date: Mon, 11 Apr 2022 16:54:27 +0100
+Subject: Add Haiku platform to setup.py
+
+
+diff --git a/setup.py b/setup.py
+index 475bc5d..8ee0d72 100644
+--- a/setup.py
++++ b/setup.py
+@@ -104,7 +104,15 @@ elif sys.platform.startswith(("darwin", "freebsd", "openbsd")):
+     libraries = ["mupdf", "mupdf-third"]
+     library_dirs = ["/usr/local/lib", "/opt/homebrew/lib"]
+     extra_link_args = []
+-
++    
++elif sys.platform.startswith("haiku"):
++    include_dirs = [
++        "/boot/system/develop/headers/mupdf",
++        "/boot/system/develop/headers/freetype2",
++    ]
++    libraries = ["mupdf"]
++    library_dirs = ["/boot/system/lib"]
++    extra_link_args = []

```

Comment:
This won't work for 32bit secondary architecture, "/boot/system/develop/headers/mupdf" vs "/boot/system/develop/headers/`x86`/mupdf"

Reply:
Works if I add the correct paths to these:
```diff
elif sys.platform.startswith("haiku"):
    include_dirs = [
        "/boot/system/develop/headers/mupdf",
+        "/boot/system/develop/headers/x86/mupdf",
        "/boot/system/develop/headers/freetype2",
+        "/boot/system/develop/headers/x86/freetype2",
    ]
    libraries = ["mupdf"]
    library_dirs = ["/boot/system/lib"]
+    library_dirs = ["/boot/system/lib/x86"]
    extra_link_args = []
```

Reply:
You might try to use pkgconfig, for instance: https://stackoverflow.com/questions/60174152/how-do-i-add-pkg-config-the-setup-py-of-a-cython-wrapper/60893447#60893447

Reply:
Nice one, will check it out

Reply:
I'll give it a go shortly...

Reply:
I've had a look into this. PyMuPDF doesn't use pkg-config for this on any other platforms so I think the patch would be simpler, shorter and easier to understand if we didn't. To this end I have pushed with the changes made by @Begasus.
I'm not opposed to using pkg-config in theory though.

Reply:
In the end the patch should be upstreamed. Why it is even hardcoded is over my head.

Reply:
I agree in principle. It's probably hardcoded just because it's simple to begin with... I haven't seen many python modules' code so I'm not sure what the normal practice is.